test/unit/helpers/ConverterHelper.ts

```diff
@@ -0,0 +1,209 @@
+        enabled: true,
+      }
+    };
+    const appConfig = await getFinalAppConfig(fakeUserConfig);
+    const promptOptions = appConfig.userConfig.promptOptions?.slidedown?.prompts[0];
+
+    t.is(promptOptions?.type, DelayedPromptType.Push);
+    t.is(promptOptions?.delay?.pageViews, 1);
+    t.is(promptOptions?.delay?.timeDelay, 0);
+    t.is(promptOptions?.autoPrompt, true);
+    t.is(promptOptions?.text.acceptButton, "Custom accept button");
+    t.is(promptOptions?.text.cancelButton, "Custom cancel button");
+    t.is(promptOptions?.text.actionMessage, "Custom message");
+  });
+
+  test("supports version 1 of config schema (converts to version 2) - slidedown type is 'push'", async t => {
+    const fakeUserConfig: AppUserConfig = {
+      appId: Random.getRandomUuid(),
+      autoRegister: false,
+      autoResubscribe: true
+    };
+
+    (fakeUserConfig as any).promptOptions = {
+      slidedown: {
+        autoPrompt: true,
+        enabled: true,
+      }
+    };
+    const appConfig = await getFinalAppConfig(fakeUserConfig);
+    const promptOptions = appConfig.userConfig.promptOptions?.slidedown?.prompts[0];
+
+    t.is(promptOptions?.type, DelayedPromptType.Push);
+    t.is(promptOptions?.delay?.pageViews, 1);
+    t.is(promptOptions?.delay?.timeDelay, 0);
+    t.is(promptOptions?.autoPrompt, true);
+    t.is(promptOptions?.text.acceptButton, "Allow");
+    t.is(promptOptions?.text.cancelButton, "Cancel");
+    t.is(promptOptions?.text.actionMessage, "We'd like to show you notifications for the latest news and updates.");
+  });
+
+  test("supports version 1 of config schema (converts to version 2) - type is 'category'", async t => {
+    const fakeUserConfig: AppUserConfig = {};
+
+    (fakeUserConfig as any).promptOptions = {
+      slidedown: {
+          autoPrompt: false,
+          enabled: true,
+          categories: {
+              positiveUpdateButton: "Howdy, yes",
+              negativeUpdateButton: "Cancelar",
+              tags: [
+                {
+                    tag: "politics",
+                    label: "Politics",
+                },
+                {
+                    tag: "usa_news",
+                    label: "USA News",
+                },
+              ]
+          }
+      }
+    };
+    const appConfig = await getFinalAppConfig(fakeUserConfig);
+    const promptOptions = appConfig.userConfig.promptOptions?.slidedown?.prompts[0];
+
+    t.is(promptOptions?.type, DelayedPromptType.Category);
+    t.is(promptOptions?.delay?.pageViews, 1);
+    t.is(promptOptions?.delay?.timeDelay, 0);
+    t.is(promptOptions?.autoPrompt, false);
+    t.is(promptOptions?.text.positiveUpdateButton, "Howdy, yes");
+    t.is(promptOptions?.text.negativeUpdateButton, "Cancelar");
+    t.is(promptOptions?.text.actionMessage, "We'd like to show you notifications for the latest news and updates.");
+  });
+
+  test("converter handles case where some settings are set in v1 format and others in v2", async t => {
```

Comment:
this test checks v2 prompt options aren't "clobbered" @jkasten2

Reply:
👍